### DISCUSSION TITLE
Add new feature

### DISCUSSION THREAD

**Discussion by HazemMohamed252** [statamic/cms#5332] [Q&A]
Hello i am new on statamic, all i want to customize th cp and add new module for example : add Notifications in sidebar with related CRUD view, controllers and routes. How can i do this please i need this so quickly

**Comment by duncanmcclean** [comment]
If you're willing to use a database, would [Runway](https://runway.duncanmcclean.com/) work for you?

**Reply by HazemMohamed252** [reply]
Yes i will use database, does you mean Runway will support me for what i want ?

**Reply by duncanmcclean** [reply]
Runway will basically make your Eloquent models look like Collections in the Control Panel, with the same interface you're used to. No need to write the controllers, routes etc

**Reply by HazemMohamed252** [reply]
After try to install it from addons i face errors : 

https://repo.packagist.org could not be fully loaded (curl error 6 while downloading https://repo.packagist.org/packages.json: getaddrinfo() thread failed to start
), package information was loaded from the local cache and may be out of date
Using version ^2.2 for doublethreedigital/runway
./composer.json has been updated
Running composer update doublethreedigital/runway --with-dependencies
> Statamic\Console\Composer\Scripts::preUpdateCmd
Loading composer repositories with package information
https://repo.packagist.org could not be fully loaded (curl error 6 while downloading https://repo.packagist.org/packages.json: getaddrinfo() thread failed to start
), package information was loaded from the local cache and may be out of date
Dependency statamic/cms is also a root requirement. Package has not been listed as an update argument, so keeping locked at old version. Use --with-all-dependencies (-W) to include root dependencies.
Updating dependencies
Nothing to modify in lock file
Writing lock file
Installing dependencies from lock file (including require-dev)
Nothing to install, update or remove
Package swiftmailer/swiftmailer is abandoned, you should avoid using it. Use symfony/mailer instead.
Generating optimized autoload files
> Illuminate\Foundation\ComposerScripts::postAutoloadDump
> @php artisan package:discover --ansi
Discovered Package: ajthinking/archetype
Discovered Package: barryvdh/laravel-debugbar
Discovered Package: doublethreedigital/runway
Discovered Package: facade/ignition
Discovered Package: fruitcake/laravel-cors
Discovered Package: intervention/image
Discovered Package: laravel/tinker
Discovered Package: nesbot/carbon
Discovered Package: nunomaduro/collision
Discovered Package: rebing/graphql-laravel
Discovered Package: statamic/cms
Discovered Package: wilderborn/partyline
Package manifest generated successfully.
> @php artisan statamic:install --ansi

   GuzzleHttp\Exception\ConnectException 

  cURL error 6: getaddrinfo() thread failed to start
 (see https://curl.haxx.se/libcurl/c/libcurl-errors.html) for https://statamic.com/api/v1/marketplace/packages/doublethreedigital/runway/v2.2.5

  at D:\develop\sta\vendor\guzzlehttp\guzzle\src\Handler\CurlFactory.php:210
    206▕         }
    207▕ 
    208▕         // Create a connection exception if it was a specific error code.
    209▕         $error = isset($connectionErrors[$easy->errno])
  ➜ 210▕             ? new ConnectException($message, $easy->request, null, $ctx)
    211▕             : new RequestException($message, $easy->request, $easy->response, null, $ctx);
    212▕ 
    213▕         return P\Create::rejectionFor($error);
    214▕     }

  1   D:\develop\sta\vendor\guzzlehttp\guzzle\src\Handler\CurlFactory.php:158
      GuzzleHttp\Handler\CurlFactory::createRejection(Object(GuzzleHttp\Handler\EasyHandle))

  2   D:\develop\sta\vendor\guzzlehttp\guzzle\src\Handler\CurlFactory.php:110
      GuzzleHttp\Handler\CurlFactory::finishError(Object(GuzzleHttp\Handler\CurlHandler), Object(GuzzleHttp\Handler\EasyHandle), Object(GuzzleHttp\Handler\CurlFactory))
Script @php artisan statamic:install --ansi handling the post-autoload-dump event returned with error code 1

**Reply by duncanmcclean** [reply]
Try installing it via the command line, not via the Control Panel.

**Reply by HazemMohamed252** [reply]
i am try to create model Order but some issues : 

Class 'App\Models\Order' not found

  at D:\develop\sta\vendor\doublethreedigital\runway\src\Resource.php:36
     32▕     {
     33▕         return $this->fluentlyGetOrSet('model')
     34▕             ->setter(function ($value) {
     35▕                 if (! $value instanceof Model) {
  ➜  36▕                     return new $value();
     37▕                 }
     38▕
     39▕                 return $value;
     40▕             })

  1   D:\develop\sta\vendor\statamic\cms\src\Support\FluentGetterSetter.php:130
      DoubleThreeDigital\Runway\Resource::DoubleThreeDigital\Runway\{closure}("App\Models\Order")

  2   D:\develop\sta\vendor\statamic\cms\src\Support\FluentGetterSetter.php:97
      Statamic\Support\FluentGetterSetter::runSetterLogic("App\Models\Order")

**Reply by HazemMohamed252** [reply]
php please runway:generate-migrations resource-handle
Generating migrations...


   DoubleThreeDigital\Runway\Exceptions\ResourceNotFound

  Runway could not find [resource-handle]. Please ensure its configured properly and you're using the correct handle.

  at D:\develop\sta\vendor\doublethreedigital\runway\src\Runway.php:82
     78▕     {
     79▕         $resource = collect(static::$resources)->get($resourceHandle);
     80▕
     81▕         if (! $resource) {
  ➜  82▕             throw new ResourceNotFound($resourceHandle);
     83▕         }
     84▕
     85▕         return $resource;
     86▕     }

  1   D:\develop\sta\vendor\doublethreedigital\runway\src\Console\Commands\GenerateMigration.php:192
      DoubleThreeDigital\Runway\Runway::findResource("resource-handle")

  2   D:\develop\sta\vendor\laravel\framework\src\Illuminate\Container\BoundMethod.php:36
      DoubleThreeDigital\Runway\Console\Commands\GenerateMigration::handle()

**Reply by duncanmcclean** [reply]
Please open a separate issue for this over on the [Runway repository](https://github.com/doublethreedigital/runway/issues) as this is no longer a Statamic issue.

**Reply by HazemMohamed252** [reply]
Thanks so much
